### DISCUSSION TITLE
Price modifiers for Line Item options

### DISCUSSION THREAD

**Discussion by nfourtythree** [craftcms/commerce#2345] [Ideas]
Ability to set-up price modifiers for variants based on line item options.

**Comment by ahausamm** [comment]
Any news on this @nfourtythree? Or still the implementation mentioned on https://github.com/craftcms/commerce/issues/1047 recommended?

**Comment by Mosnar** [comment]
I just wanted to leave a note that Shopify does variable products in a surprisingly simple & elegant way that I think is worth keeping in mind for Commerce.

The gist of it is that Shopify lets you define attributes and values per-product - as you define the variables, a table is generated on the page of all possible combinations. You can make basic edits to this table at first, but after you save the product, Shopify generates these variants and you get a very slick & responsive interface for filtering, bulk editing, and managing stock levels for all of these variants. I can't overstate how important this is for clients with many stock codes. Managing things like this in the current Commerce UI is certainly possible; however, definitely not practical.

To boil it down, these are my key take-aways from the Shopify implementation:
- Variable products often results in a ton of data - fast and responsive UI with Excel-like efficiency is a must
- They _could_ have opted to lock product attributes down to predefined options (e.g. Size, Color, etc), but instead opted for leaving it open ended for authors.
- Just because we know how to dynamically generate variants doesn't mean these variants don't have data that needs to be user managed. Consider for example a shirt with colors, sizes, and styles. A store manager should be able to add a style option without needing to spend 30 minutes ensuring all new variants exist manually or worse, waiting for a developer to build and run a Feed Me import.

**Comment by nfourtythree** [comment]
Thanks for the input @Mosnar 

The idea behind this feature is more along the lines of adding $5 if you select "gift wrapped". 

What you have mentioned is more to do with product attributes which we have an feature request for and is on our radar #1060

Thanks!

**Comment by billmn** [comment]
> The idea behind this feature is more along the lines of adding $5 if you select "gift wrapped".

@nfourtythree While waiting for this feature, what would be the recommended way to accomplish that in Commerce 3.2?

Use a `custom` adjuster type even if docs specifies that it is deprecated and will be removed in Commerce 4? https://craftcms.com/docs/commerce/3.x/extend/adjusters.html#writing-an-adjuster

Thank you!

**Comment by nfourtythree** [comment]
Hi @billmn 

The most common way to modify a line item's price is to use the `LineItems::EVENT_POPULATE_LINE_ITEM` event.

You can see an example of this in [the following Stack Exchange post](https://craftcms.stackexchange.com/questions/27477/user-determined-price-for-a-product-variant)

Thanks

**Reply by nilsenpaul** [reply]
Hi @nfourtythree, I recently implemented this for a giftcard product, and ran into getting an error on trying to complete the order (along the lines of "your order price changed"). I went with a negative Discount adjuster now, but the word 'DISCOUNT' on the order detail page seems a bit ... odd.

**Reply by nilsenpaul** [reply]
But then again, after what I've heard about how this will eventually work in Commerce 4, I've just replaced DISCOUNT by ADDITION for now.
